### PR TITLE
Feat: Adds on demand taxi routes

### LINKED ISSUE #350: Booking On-Demand Transport (Taxi)
In Herrenberg after 9pm some busses are replaced with Taxis that you need to book in advance.

For example, when you want to travel from Nufringen to Gültstein at 10 pm, you will receive something that looks like a normal bus route: https://tinyurl.com/yds6qdkz

Screenshot from 2021-05-31 09-26-21

## Flex mode

However, you can also book a special mode that will take you directly to your door (if it's within 300m of a bus stop). We want to show this mode in a separate "tile" next to the cycling, p+r ... modes. 
This is the "Taxi" Icon that we are using combined with the public transit icon for the new flex mode tile: 
[TaxiIcon.zip](https://github.com/stadtnavi/digitransit-ui/files/6605014/TaxiIcon.zip)

Screenshot from 2021-05-31 10-04-01

The icon should be a car with a bus in the background.

In this tile, you need to add the modes `FLEX_DIRECT`, `FLEX_EGRESS`. The result should then look like this.

Screenshot from 2021-05-31 10-09-40

## Booking info

Also in both cases you must get additional data from the GraphQL API about the following booking requirements:

- booking telephone number and URL
- a warning about having to book ahead of time

We have added new GraphQL properties for the booking info and with the following query you can get the message, telephone number and URL.

URL: https://api.dev.stadtnavi.eu/otp/routers/default/index/graphql

```
query{
  plan(
    date: "2021-06-02"
    time: "22:30"
    from: { lat:  48.6226, lon: 8.8879 }
    to: { lat: 48.5755, lon: 8.8795 }
    transportModes: [
      { mode: RAIL },
      { mode: FLEX, qualifier: EGRESS },
      { mode: FLEX, qualifier: DIRECT },
      { mode: WALK },
    ],
    maxWalkDistance: 5000
  ) {
    itineraries {
      legs {
        mode
        rentedBike
        distance
        startTime
        endTime
        pickupBookingInfo {
          message
          contactInfo {
            phoneNumber
            infoUrl
            bookingUrl
          }
        }
        route {
          url
          mode
          shortName
        }
        trip {
          gtfsId
          tripShortName
        }
        from {
          name
          bikeRentalStation {
            name
            id
          }
        }
        to {
          name
          bikeRentalStation {
            name
            id
          }
        }
      }
    }
  }
}
```

The new part is:

```
 pickupBookingInfo {
          message
          contactInfo {
            phoneNumber
            infoUrl
            bookingUrl
          }
        }
```

### PR DESCRIPTION
**Pull request by andreashelms**
## Proposed Changes

image
image

## Review

  - Read and verify the code changes
  - Test the functionality by running the UI locally with all popular browsers available in your platform

closes #350

### CODE REVIEW

File: app/component/TransitLeg.js [hunk 1]
```diff
@@ -393,13 +394,67 @@ class TransitLeg extends React.Component {
                   color={leg.route ? `#${leg.route.color}` : 'currentColor'}
                   text={leg.route && leg.route.shortName}
                   realtime={false}
+                  icon={
+                    leg.route.shortName.startsWith('RT')
```
Comment (andreashelms) [inline app/component/TransitLeg.js]: @leonardehrenfried Can you think of any other possibility to check whether a leg is of type "Ruftaxi"? Mode is always bus for Ruftaxi.

Comment (leonardehrenfried) [reply app/component/TransitLeg.js]: In theory you there is `route.type` which can have lots of values depending on the specific type: https://developers.google.com/transit/gtfs/reference/extended-route-types

I'm not sure if we are setting this correctly in the input feed. @derhuerst can you clarify?

Comment (leonardehrenfried) [reply app/component/TransitLeg.js]: It seems that type 1500 (Taxi service) would be the most appropriate route type.

Comment (leonardehrenfried) [reply app/component/TransitLeg.js]: I'm not finding a place where the route type is changed from bus (3) to something else: https://github.com/derhuerst/generate-herrenberg-gtfs-flex/search?q=type

**Comment by hbruch** [comment]
In the GTFS-data, the route type is 715 (OnDemand)

**Comment by derhuerst** [comment]
I turned out to be a change in the underlying/previous processing step. I've adapted `generate-herrenberg-gtfs-flex` so that it always sets `route_type: 715` for all specified routes.

@andreashelms Once @hbruch has ran another `hbg3.merged.with_flex.gtfs.zip` feed build with it, you should be able to use `route_type === '715'` as a check.

**Comment by leonardehrenfried** [comment]
I will let you know when this is the case.

**Pull request closed without merging**